InAppUtils/InAppUtils.m

```diff
@@ -199,6 +199,37 @@ - (void)dealloc
     [[SKPaymentQueue defaultQueue] removeTransactionObserver:self];
 }
 
+RCT_EXPORT_METHOD(paidForApp:(RCTResponseSenderBlock)callback)
+{
+    NSString *restoreRequest = @"paidForAppRequest";
+    _callbacks[RCTKeyForInstance(restoreRequest)] = callback;
+    SKReceiptRefreshRequest* request = [[SKReceiptRefreshRequest alloc] initWithReceiptProperties:nil];
+    request.delegate = self;
+    [request start];
+}
+
+-(void)requestDidFinish:(SKRequest*)request{
+    if([request isKindOfClass:[SKReceiptRefreshRequest class]]){
+        NSLog(@"YES, You purchased this app");
+        NSString *key = RCTKeyForInstance(@"paidForAppRequest");
+        RCTResponseSenderBlock callback = _callbacks[key];
+        if (callback) {
+            callback(@[@"paid_for_app"]);
+            [_callbacks removeObjectForKey:key];
+        }
+    }
+}
+
+- (void)request:(SKRequest*)request didFailWithError:(NSError *)error{
+    NSLog(@"NO, you need to buy it");
+    NSString *key = RCTKeyForInstance(@"paidForAppRequest");
```

Comment:
This delegate method will catch all request failures. even the ones for buying something. Eventually, you are calling a wrong callback when request fails.